### ISSUE TITLE
question: ignoring errors from specific  files

### ISSUE THREAD

**Issue by TkDodo** [webpro/knip#340]
I'm getting some errors that I don't really care about from `.d.ts` files. For example, we have:

```ts
declare module '*.svg' {
    import type { FC, SVGProps } from 'react'

    export const ReactComponent: FC<SVGProps<SVGSVGElement> & { title?: string }>

    const src: string
    export default src
}
```

in a `.d.ts` file because we use svgr. knip shows both `ReactComponent` and the default export as unused (note: this might be a bug, not sure?)

Now I added: `**/*.d.ts` to the `ignore` config of knip and expected those errors to go away, but they don't 🤔.

So the question is: is `ignore` the right config to use for this? If this should work, I'll try to create a reproduction. Thanks 🙏

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.40.2. See [Release 2.40.2](https://github.com/webpro/knip/releases/tag/2.40.2) for release notes._

**Comment by webpro** [comment]
This inspired me to look at only top-level nodes for exports, which is also a tiny perf win. Perhaps can do some of small wins like this in the future, but unfortunately Knip needs to scan the full AST at least for dynamic `import()` calls.
